azuread/data_application.go

```diff
@@ -18,19 +18,27 @@ func dataApplication() *schema.Resource {
 
 		Schema: map[string]*schema.Schema{
 			"object_id": {
-				Type:          schema.TypeString,
-				Optional:      true,
-				Computed:      true,
-				ValidateFunc:  validate.UUID,
-				ConflictsWith: []string{"name"},
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ExactlyOneOf: []string{"name", "application_id"},
```

Comment:
I think you'll need object_id in here?
```suggestion
				ExactlyOneOf: []string{"name", "application_id", "object_id"},
```

Reply:
Ah yes, thanks!